### PR TITLE
Generate weakest known-bits precondition disjunction.

### PR DESCRIPTION
**Pull request by manasij7479**
Limited to one variable functions now.

### CODE REVIEW

**Comment by regehr** [comment]
is this a draft, or something we should merge?

File: lib/Extractor/Solver.cpp [hunk 1]
```diff
@@ -262,6 +262,154 @@ class BaseSolver : public Solver {
...
+          llvm::outs() << "Transformation is not valid for any input.\n";
+          return EC;
+        } else {
+          FoundWeakest = true;
+          llvm::outs() << "Exhausted search space.\n";
+          break;
+        }
+      }
+
+      for (unsigned I=0; I< W; I++) {
```
Comment (regehr) [inline lib/Extractor/Solver.cpp:348]: avoid sloppy whitespace like I< W

File: test/Infer/ap/kb.opt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+%0:i8 = var
+%1 = add %0, 1
+%2 = or %1, 7
+infer %2
+%3 = add %0, 2
+%4 = or %3, 7
+result %4
+;CHECK: xxxxxxx1
+;CHECK: xxxxx0xx
+;CHECK: xxxxxx0x
```
Comment (regehr) [inline test/Infer/ap/kb.opt:37]: I don't understand the multiple CHECK lines, does your code print multiple solutions if these exist?

File: test/Infer/ap/kb.opt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+%0:i8 = var (knownBits=xxxxxxx0)
+%1:i8 = add 1:i8, %0
+%2:i8 = or 7:i8, %1
+infer %2
+%3:i8 = add 2:i8, %0
+%4:i8 = or 7:i8, %3
+result %4
+;CHECK: xxxxx0xx
+;CHECK: xxxxxx0x
+
```
Comment (regehr) [inline test/Infer/ap/kb.opt:48]: how about add one test case where the result is all unknown bits?

File: test/Infer/ap/kb.opt (hunk 2 above)
Comment (manasij7479) [reply test/Infer/ap/kb.opt:37]: Yes, it prints all the solutions.
So the weakest precondition is the disjunction of the results.

File: test/Infer/ap/kb.opt (hunk 3 above)
Comment (manasij7479) [reply test/Infer/ap/kb.opt:48]: All unknown exits early saying the optimization is already valid.
I'll add a test checking for that.

**Pull request merged**